docs/manual/constant-value-checker.tex

```diff
@@ -251,6 +251,39 @@ \section{Warnings\label{value-checker-warnings}}
 
 \end{itemize}
 \end{sloppypar}
+
+
+\section{Overflow\label{value-checker-overflow}}
+
+The Constant Value Checker normally takes Java's normal overflow rules into account when
+computing the possible values of expressions.
+The \code{-AignoreRangeOverflow} command-line option
+makes it ignore the possibility of overflow
+for range annotations (\<@IntRange>, for instance), making them more precise.
+You can supply the \code{-AignoreRangeOverflow} command-line option
+when you are confident that no overflow will occur in the program (for
```

Comment:
I'm not sure it would always be possible to write upper bound annotations instead of using `-AignoreRangeOverflow`. It's possible that a programmer might not know the precise upper bound but are still confident that there will not be overflow; this is the assumption that the Index Checker makes.

From a design perspective, I view this as separation of concerns - checking that there is no overflow is one thing, and doing constant propagation a la the value checker is a different thing. If they're tightly coupled (as they are without the `-AignoreRangeOverflow` flag) then there is no way to get one without the other, which is what we need in the Index Checker.